src/main/java/edu/umich/its/spe/SPEUtils.java

```diff
@@ -19,8 +49,110 @@ static protected String safeGetPropertyValue(HashMap<String, String> testpropert
+		}
+
+		M_log.debug("normalizeStringTimestamp: without fractional seconds {}",stringTimestamp);
+
+		// Make sure has explicit timezone information.  Either it has an offset or ends with 'Z' (UTC).
+
+		Matcher mo = p_offset.matcher(stringTimestamp);
+
+		if (mo.find()) {
+			// found this has an offset, so make sure it has the embedded colon.
+			if (! ":".equals(String.valueOf(stringTimestamp.substring(stringTimestamp.length()-3,
+					stringTimestamp.length()-2)))) {
+				stringTimestamp = new StringBuffer(stringTimestamp).insert(stringTimestamp.length()-2,":").toString();
+			}
+			M_log.debug("normalizeStringTimestamp: standarized colon in offset {}",stringTimestamp);
+		}
+		else {
+			// no offset so make sure there is (only 1) Z at the end.
+			if (! "Z".equalsIgnoreCase(String.valueOf(stringTimestamp.charAt(stringTimestamp.length()-1)))) {
+				M_log.debug("normalizeStringTimestamp: adding Z to {}",stringTimestamp);
+				stringTimestamp += "Z";
+			}
+		}
+
+		// make sure the ISO 8601 T is in there.
+		stringTimestamp = stringTimestamp.replaceFirst( " ", "T" );
+
+		M_log.debug("normalizeStringTimestamp output: {}",stringTimestamp);
+		return stringTimestamp;
+	}
+
+	// Returns as plain date time string assumed to be in UTC.  There is no offset or Z.
+	public static String formatTimestampInstantToImplicitUTC(Instant timeStampInstant) {
+		// by default this converts to UTC in ISO8601.  We format without time zone information.
+		// User must assume this is in UTC.
+		return timeStampInstant.toString().replace("Z","");
+	}
+
+	// Get a long value from a string use default if the string is null or empty.
+	public static long longFromStringWithDefault(String longString, Long defaultValue) {
+		long longValue = (longString == null ? defaultValue : Long.parseLong(longString));
+		M_log.debug("longFromString: {} {}",longString,longValue);
+		return longValue;
+	}
+
+	// get an existing offset time and return a new string in UTC that is one second later.
+	public static Instant generateNewQueryTime(Instant priorUpdateTime) {
+
+		Instant newQueryTime = priorUpdateTime.plusSeconds(1);
+		M_log.debug("generating new query time: original: {} new {}",priorUpdateTime,newQueryTime);
+
+		return newQueryTime;
+	}
+
+	public static Instant convertTimeStampStringToInstant(String timestamp) {
+
+		// This puts the string in a standard format that should be parsable.
+		String s = SPEUtils.normalizeStringTimestamp(timestamp);
+
+		Matcher mo = p_offset.matcher(timestamp);
+
+		if (mo.find()) {
+			M_log.debug("found offset in timestamp: {} so trying OffsetDateTime",timestamp);
+			try {
+				OffsetDateTime odt = OffsetDateTime.parse(s);
+				M_log.debug("Offset date time for: {} is {}",s,odt);
+				//return Instant.parse(s,DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+				return odt.toInstant();
+			} catch(DateTimeParseException exp) {
+				M_log.info("OffsetDateTime.parse: Catch {} for timestamp: {}",exp,s);
```

Comment:
Shouldn't this be `error` instead

Reply:
It's not an error since it may be that the next step of parsing as an Instant will work fine.  The only error would be if every one of the possible parses fails.  That case is noted as an error.